src/Test/MakerTestDetails.php

```diff
@@ -64,6 +68,47 @@ public function setFixtureFilesPath(string $fixtureFilesPath): self
         return $this;
     }
 
+    public function changeRootNamespace(string $rootNamespace): self
+    {
+        $rootNamespace = trim($rootNamespace, '\\');
+
+        // to bypass read before flush issue
+        $this->snapshotSuffix = $rootNamespace;
+
+        return $this
+            ->createFile('config/packages/dev/maker.yaml', "maker:\n    root_namespace: Custom\n")
+            ->addReplacement(
+                'composer.json',
+                '"App\\\\": "src/"',
+                '"'.$rootNamespace.'\\\\": "src/"'
+            )
+            ->addReplacement(
+                'src/Kernel.php',
+                'namespace App',
+                'namespace '.$rootNamespace
+            )
+            ->addReplacement(
+                'bin/console',
+                'use App\\Kernel',
+                'use '.$rootNamespace.'\\Kernel'
+            )
+            ->addReplacement(
+                'public/index.php',
+                'use App\\Kernel',
+                'use '.$rootNamespace.'\\Kernel'
+            )
+            ->addReplacement(
+                'config/services.yaml',
+                'App\\',
+                $rootNamespace.'\\'
+            )
+            ->addReplacement(
+                'phpunit.xml.dist',
+                '<env name="KERNEL_CLASS" value="App\\Kernel" />',
+                '<env name="KERNEL_CLASS" value="'.$rootNamespace.'\\Kernel" />'
+            );
```

Comment:
I don't know if we need to do all of this. I should be enough to create the `maker.yaml` file with the config AND add one new PSR4 namespace autoload to `composer.json`. I don't think we need to completely eliminate and convert the rest of the app from the `App\` namespace.

Reply:
At first I did so. But such solution has issue: tests pass when it mustn't because autoloader. After debugging tests we can remove it. But I think complete converting namespace is better because it is case for we are doing all of this :smile: